hazelcast/src/main/java/com/hazelcast/internal/affinity/impl/TypedAffinityEnforcerInterceptor.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.affinity.impl;
+
+import com.hazelcast.internal.affinity.ThreadAffinityParamsHelper;
+import com.hazelcast.internal.affinity.ThreadAffinity;
+import com.hazelcast.internal.affinity.ThreadAffinityInterceptor;
+import com.hazelcast.logging.ILogger;
+import net.openhft.affinity.AffinityLock;
+
+import java.util.Set;
+
+class TypedAffinityEnforcerInterceptor
```

Comment:
IMHO too abstract. How many implementations do we have? Just make it as concrete as possible.

Reply:
The interceptor has a NOOP impl as well, thus the interface.

Reply:
IMHO that is making the design a lot more complicated than is needed. A single concrete class is sufficient to do it all.

Reply:
Done